adoc/chapters/cuda_backend.adoc

```diff
@@ -0,0 +1,551 @@
+functions of classes [code]#platform#, [code]#context#, [code]#device#, 
+[code]#queue#, [code]#event# and [code]#kernel#.
+
+[[sub:cuda:application_interoperability]]
+=== Application Interoperability
+
+This section describes the supported application interoperability between the
+SYCL generic programming model and CUDA.
+
+The CUDA backend supports application interoperability for creating a SYCL
+object from a native CUDA object for [code]#device#, [code]#context#,
+[code]#queue# and [code]#event#. This kind of interoperability is not supported
+for [code]#platform#, [code]#buffer#, [code]#kernel#, [code]#kernel_bundle#,
+[code]#device_image#, [code]#sampled_image# and [code]#unsampled_image# are not
+supported.
+ 
+The CUDA backend supports application interoperability for retrieving a native
+CUDA object from a SYCL object for [code]#device#, [code]#context#,
+[code]#queue# and [code]#event#. This kind of interoperability is not supported
+for [code]#platform#, [code]#buffer#, [code]#kernel#, [code]#kernel_bundle#,
+[code]#device_image#, [code]#sampled_image# and [code]#unsampled_image# are not
+supported.
+
+[[table.cuda.appinterop.nativeobjects]]
+.Types of native backend objects application interoperability
+[width="100%",options="header",cols="20%,20%,20%,40%"]
+|====
+| [code]#SyclType# | [code]#backend_input_t<backend::cuda, SyclType># | [code]#backend_return_t<backend::cuda, SyclType># | Description
+| [code]#device# | [code]#CUdevice# | [code]#CUdevice# | A SYCL [code]#device# created from a [code]#CUdevice# will return a copy of an existing device with that native [code]#CUdevice#. The [code]#CUdevice# retireved from a SYCL[code]#device# will be the native  [code]#CUdevice# associated with it.
+| [code]#context# | [code]#CUcontext# | [code]#std::vector<CUcontext># | A SYCL [code]#context# created from a single [code]#CUcontext# will encapsulate that context. All [code]#CUcontext#s associated with a SYCL [code]#context# are retireved from a SYCL [code]#context#. The native [code]#CUcontext#s may contain onr or more CUDA primary contexts.
+| [code]#queue# | [code]#CUstream# | [code]#CUstream# | A SYCL [code]#queue# created from a [code]#CUstream# will encapsulate that stream and will have the [code]#property::queue::in_order# property . A single [code]#CUstream# is retrieved from a SYCL[code]#queue#, and before it is returned all native [code]#CUstream#s associated with the SYCL [code]#queue# synchronize with the calling thread.
+| [code]#event# | [code]#CUevent# | [code]#CUevent# | A SYCL [code]#event# created from a [code]#CUstream# will encapsulate that event. A single [code]#CUevent# is potentially retrieved from a SYCL [code]#event# if there is a valid native [code]#CUevent# associated with it, otherwise [code]#nullptr# is returned instead. The CUDA backend-specific free function [code]#cuda::has_native_event# can be used to query whether the SYCL [code]#event# has a valid native [code]#CUevent# associated with it.
```

Comment:
```suggestion
| [code]#event# | [code]#CUevent# | [code]#CUevent# | A SYCL [code]#event# created from a [code]#CUevent# will encapsulate that event. A single [code]#CUevent# is potentially retrieved from a SYCL [code]#event# if there is a valid native [code]#CUevent# associated with it, otherwise [code]#nullptr# is returned instead. The CUDA backend-specific free function [code]#cuda::has_native_event# can be used to query whether the SYCL [code]#event# has a valid native [code]#CUevent# associated with it.
```